src/Symfony/Component/Console/Tests/Input/ArgvInputTest.php

```diff
@@ -455,4 +455,33 @@ public function testParseOptionWithValueOptionalGivenEmptyAndOptionalArgument()
         $this->assertEquals(['foo' => '0'], $input->getOptions(), '->parse() parses optional options with empty value as null');
         $this->assertEquals(['name' => 'bar'], $input->getArguments(), '->parse() parses optional arguments');
     }
+
+    /**
+     * @dataProvider provideParameterToConsume
+     *
+     * @group argv
+     */
+    public function testParameterIsConsumed($argv, $key, $onlyParams, $consumed, $expected)
+    {
+        $input = new ArgvInput($argv);
+        $this->assertEquals($expected, $input->getParameterOption($key, false, $onlyParams, $consumed), 'The parameter has been consumed');
+    }
+
+    public function provideParameterToConsume()
+    {
+        return [
+            [['app/console', 'foo:bar'], '-e', false, false, ''],
+            [['app/console', 'foo:bar'], '-e', false, true, ''],
+            [['app/console', 'foo:bar', '-e', 'dev'], '-e', false, false, 'dev'],
+            [['app/console', 'foo:bar', '-e', 'dev'], '-e', false, true, ''],
+            [['app/console', 'foo:bar', '-e', 'dev', '-f', 'bar'], ['-e', '-f'], false, false, 'dev'],
+            [['app/console', 'foo:bar', '-e', 'dev', '-f', 'bar'], ['-e', '-f'], false, true, ''],
```

Comment:
when consume is true, the first call should return true, and the next ones should return false, because the first call "consumed" the info